src/reader/buffered_reader.rs

```diff
@@ -90,11 +90,7 @@ macro_rules! impl_buffered_source {
             }
             *position += read;
 
-            if read == 0 {
-                Ok(None)
-            } else {
-                Ok(Some(&buf[start..]))
-            }
+            Ok((&buf[start..], done))
```

Comment:
`done` ought to be renamed to `found`

Reply:
Actually, all that code will be removed soon, because I rewrite it, so not necessary to polish it. That changes was made only because I plan to make tests first to know that rewriting did not break anything